stroeer/core/v1/shared.proto

```diff
@@ -6,31 +6,41 @@ option go_package = "github.com/stroeer/tapir/go/stroeer/core/v1;core";
 option java_multiple_files = true;
 option java_package = "de.stroeer.core.v1";
 
-
-// References to get access to the article on different technical platforms like
-// canonical web or AMP.
-message References {
-  // URL path for this article e.g. /section/id_$ID/title.html (required).
-  string path = 1;
-
-  // Canonical URL of the article, may differ if external, e.g. https://www.giga.de/external.html
-  // (required).
-  string canonical = 2;
-
-  // AMP URL of the article.
-  string amp = 3;
+// Represents a single Reference or Link incl. all needed attributes to built
+// an anchor element.
+message Reference {
+  // The type is only needed for filtering multiple links of the same type in
+  // list of references.
+  //
+  // Example entries:
+  // * `title`
+  // * `navigation`
+  // * `sub_navigation`
+  string type = 1;
+
+  // The text or label of the link.
+  string text = 2;
+
+  // Contains all optional attributes of a link.
+  // * `href`
+  // * `target`
+  // * `rel`
+  // * `flag:internal`
+  //
+  // Clients must be resilient to unknown or missing entry sets.
+  map<string, string> fields = 3;
 }
 
-// Hierarchical section information of the article till root.
+// Hierarchically structured references. Representing the section tree or
+// navigation menus.
 //
-// For example:  /politik/deutschland/ -> /politik/ -> /
-message SectionTree {
-  // Section path of the article, e.g. /politik/deutschland/ (required).
-  string path = 1;
-
-  // Label of the section path, e.g. "Politik".
-  string label = 2;
-
-  // Parent section. Usually points to the parent section, if there is none, e.g. when in `root` this value is `null`
-  SectionTree parent = 3;
+// For example:
+// `/` -> `/sport/` -> `/sport/fussball/`
+// [?] The name `Category` is used here to align namings with the CMS.
+message Category {
```

Comment:
IMO we shouldn't refer to the CMS for naming but decide which name is most suitable for our API and it's clients. We currently mix `Category` and `section_tree` in message name, comments and attribute names which can be confusing. 

I kind of liked the `tree` inside the naming since it represents a tree like structure.

Reply:
Thats true for the actual section tree but not for an navigation. So we need a name what can describe an section tree item and a navigation item

Reply:
We could pull the fields of `Reference` up here and rename ~Category~ into `Reference`:

```protobuf
message Reference {
  string type = 1;
  string text = 2;
  map<string, string> fields = 3;
 repeated Reference children =4;
}
```

Reply:
haha true :D

Reply:
```proto
message Reference {
  string type = 1;
  string label = 2;
  string href = 3;
  map<string, string> fields = 4;
  repeated Reference children = 5;
}

message Article {
  // ....
  repeated Reference references = 1;
  Reference section_tree = 2;
}

message Section {
  // ....
  repeated Reference references = 1;
  Reference section_tree = 2;
}
```